packages/web/src/new/scenes/Root/Root.tsx

```diff
@@ -9,8 +12,18 @@ interface Props {
   children: React.ReactNode;
 }
 
-export const Root: FC<Props> = ({ children }) => {
-  useViewModel(RootViewModel);
+export const Root: FC<Props> = observer(({ children }) => {
+  const rootViewModel = useViewModel(RootViewModel);
+  const location = useLocation();
+  const isAuth = location.pathname === '/' || location.pathname === '/trial';
+
+  if (!rootViewModel.walletModel.connected && !isAuth) {
+    return <Redirect to={'/trial'} />;
+  }
+
+  if (rootViewModel.walletModel.connected && isAuth) {
```

Comment:
I think you should use reaction in maybe Root ViewModel (you can use `reaction` or `when` for example) where you can call push of `locationService`

Reply:
I did in the component because I also need to access `react-router` to govern routing. Additionally, as mobx docs states, we should minimize usage of reactions when it is possible.